distributed/executor.py

```diff
@@ -974,6 +992,45 @@ def cancel(self, futures):
         """
         return sync(self.loop, self._cancel, futures)
 
+    def publish_dataset(self, data, name):
+        """
+        Store a named reference to the data in the scheduler for other executors
+
+        Parameters
+        ----------
+        data: list of futures or dask collection
+        name: str
+            handle by which the stored data should be known
+
+        Returns
+        -------
+        None
+        """
+        keys = [tokey(f.key) for f in futures_of(data)]
+        self._send_to_scheduler({'op': 'publish-data', 'keys': keys,
```

Comment:
Ah, I also recommend using `yield self.scheduler.publish_data(keys=keys, data=..., name=name)` instead.

There are two channels from the Executor to the Scheduler.  One is the `Executor.scheduler` rpc object for general, infrequent, synchronous communication.  The other is the `Executor.scheduler_stream` object for rapid high-volume communications (like 10000 calls to submit).  

In this case we probably want to return only after we've actually sent the data up and we're not in a high message volume case, so we probably want to use the non-specialized `Executor.scheduler` solution.  This will also require moving the `publish-data` route in the scheduler to the normal handlers dict